girder/cumulus/plugin_tests/volume_test.py

```diff
@@ -692,3 +707,58 @@ def test_get_status(self, get_ec2_client):
             u'status': u'in-use'
         }
         self.assertEqual(r.json, expected, 'Unexpected status')
+
+    @mock.patch('girder.plugins.cumulus.volume.get_ec2_client')
+    @mock.patch('cumulus.ansible.tasks.volume.create_volume.delay')
+    def test_log(self, get_ec2_client, create_volume):
+        volume_id = 'vol-1'
+        ec2_client = get_ec2_client.return_value
+        ec2_client.create_volume.return_value = {
+            'VolumeId': volume_id
+        }
+
+        body = {
+            'name': 'test',
+            'size': 20,
+            'zone': 'us-west-2a',
+            'type': 'ebs',
+            'profileId': self._profile_id
+        }
+
+        r = self.request('/volumes', method='POST',
+                         type='application/json', body=json.dumps(body),
+                         user=self._user)
+        self.assertStatus(r, 201)
+        volume_id = str(r.json['_id'])
+
+        log_entry = {
+            'msg': 'Some message'
+        }
+
+        r = self.request('/volumes/546a1844ff34c70456111185/log', method='GET',
+                         user=self._user)
+        self.assertStatus(r, 404)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='POST',
+                         type='application/json', body=json.dumps(log_entry), user=self._user)
+        self.assertStatusOk(r)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='GET',
+                         user=self._user)
+        self.assertStatusOk(r)
+        expected_log = {u'log': [{u'msg': u'Some message'}]}
+        self.assertEqual(r.json, expected_log)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='POST',
+                         type='application/json', body=json.dumps(log_entry), user=self._user)
+        self.assertStatusOk(r)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='GET',
+                         user=self._user)
+        self.assertStatusOk(r)
+        self.assertEqual(len(r.json['log']), 2)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='GET',
+                         params={'offset': 1}, user=self._user)
+        self.assertStatusOk(r)
+        self.assertEqual(len(r.json['log']), 1)
```

Comment:
We should also add a test for getting the log of a newly created volume, I see the the following error: 

```
{
  "message": "KeyError: KeyError('log',)",
  "trace": [
    [
      "/opt/hpccloud/girder/girder/api/rest.py",
      489,
      "endpointDecorator",
      "val = fun(self, args, kwargs)"
    ],
    [
      "/opt/hpccloud/girder/girder/api/rest.py",
      1024,
      "GET",
      "return self.handleRoute('GET', path, params)"
    ],
```

I think the issue is that we need to init log to [] [here](https://github.com/Kitware/cumulus/blob/volume-logs/girder/cumulus/server/models/volume.py#L68-L77)